File: programmers/lv_2/이진_변환_반복하기.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[2];
        
        while (true) {
            answer[0]++;
            for (int i = 0; i < s.length(); i++) {
                if (s.charAt(i) == '0') answer[1]++;
            }
            
            s = s.replace("0", "");

            int len = s.length();
            s = Integer.toBinaryString(len);
            if (s.equals("1")) break;
        }
        
        return answer;
    }
}

File: programmers/lv_2/이진_변환_반복하기.py
def solution(s):
    answer = [0, 0]
    
    while True:
        answer[0] += 1
        for c in s:
            if c == '0':
                answer[1] += 1
        s = s.replace('0', '')        

        leng = len(s)

        s = format(leng, 'b')
        if s == '1':
            break
    
    return answer